doc/design/cluster_train/save_model.md

```diff
@@ -0,0 +1,97 @@
+# Design Doc: Save Model
+
+## Overview
+
+The model is the output of the training process. There are two
+ways from which user can obtain a model:
+
+- Save model triggered by user code: user code asks PaddlePaddle to
+  save a model.
+- Convert model from the snapshot: model being converted from
+  pservers' periodic snapshot. In this way, the user can cancel a job
+  at any time, and still have a relatively fresh model (we snapshot
+  around every 5 minutes).
```

Comment:
we can overemphasize `snapshot` with  `model snapshot` here, otherwise, someone may confuse with the `checkpoint` if he doesn't  take a look at the checkpoint.